sgxs-loaders/src/enclaveapi/mod.rs

```diff
@@ -0,0 +1,268 @@
+use std::{mem, ptr};
+use std::sync::Arc;
+use std::io::{Error as IoError, Result as IoResult, ErrorKind};
+use winapi::um::memoryapi::VirtualFree;
+use winapi::um::processthreadsapi::GetCurrentProcess;
+use winapi::um::winnt::{HANDLE, ENCLAVE_INIT_INFO_SGX, MEM_RELEASE, PAGE_ENCLAVE_THREAD_CONTROL, PAGE_ENCLAVE_UNVALIDATED,
+                        PAGE_READONLY, PAGE_READWRITE, PAGE_EXECUTE, PAGE_EXECUTE_READ, PAGE_EXECUTE_READWRITE, ENCLAVE_TYPE_SGX};
+use winapi::um::enclaveapi::{CreateEnclave, InitializeEnclave, IsEnclaveTypeSupported, LoadEnclaveData};
+use winapi::_core::ffi::c_void;
+use abi::{Attributes, Einittoken, ErrorCode, Miscselect, PageType, SecinfoFlags, Secs, Sigstruct};
+use sgxs_crate::sgxs::{MeasEAdd, MeasECreate, PageChunks, SgxsRead};
+use sgxs_crate::einittoken::EinittokenProvider;
+use sgxs_crate::loader;
+use generic::{self, EinittokenError, EnclaveLoad, Mapping};
+use crate::{MappingInfo, Tcs};
+
+#[derive(Fail, Debug)]
+pub enum SgxIoctlError {
+    #[fail(display = "I/O ctl failed.")]
+    Io(#[cause] IoError),
+    #[fail(display = "The SGX instruction returned an error: {:?}.", _0)]
+    Ret(ErrorCode),
+}
+
+#[derive(Fail, Debug)]
+pub enum Error {
+    #[fail(display = "Failed to map enclave into memory.")]
+    Map(#[cause] IoError),
+    #[fail(display = "Failed to call CreateEnclave.")]
+    Create(#[cause] IoError),
+    #[fail(display = "Failed to call LoadEnclaveData.")]
+    Add(#[cause] IoError),
+    #[fail(display = "Failed to call InitializeEnclave.")]
+    Init(#[cause] SgxIoctlError),
+}
+
+impl EinittokenError for Error {
+    fn is_einittoken_error(&self) -> bool {
+        use self::Error::Init;
+        use self::SgxIoctlError::Ret;
+        match self {
+            &Init(Ret(ErrorCode::InvalidEinitToken)) |
+            &Init(Ret(ErrorCode::InvalidCpusvn)) |
+            &Init(Ret(ErrorCode::InvalidAttribute)) | // InvalidEinitAttribute according to PR, but does not exist.
+            &Init(Ret(ErrorCode::InvalidMeasurement)) => true,
+            _ => false,
+        }
+    }
+}
+
+impl EnclaveLoad for WinInnerLibrary {
+    type Error = Error;
+
+    fn new(
+        device: Arc<WinInnerLibrary>,
+        ecreate: MeasECreate,
+        attributes: Attributes,
+        miscselect: Miscselect,
+    ) -> Result<Mapping<Self>, Self::Error> {
+        let secs = Secs {
+            size: ecreate.size,
+            ssaframesize: ecreate.ssaframesize,
+            miscselect,
+            attributes,
+            ..Default::default()
+        };
+
+        let curhandle: HANDLE = unsafe {GetCurrentProcess()};
```

Comment:
Just do the call to `GetCurrentProcess()` directly as a function argument of `CreateEnclave`. You can do this everywhere.